lib/msf/core/post/windows/dotnet.rb

```diff
@@ -0,0 +1,99 @@
+# -*- coding: binary -*-
+require 'msf/core/post/common'
+require 'msf/core/post/windows/registry'
+
+module Msf
+class Post
+module Windows
+
+#
+# This module serves to hold methods related to .NET framework
+#
+module Dotnet
+  include ::Msf::Post::Common
+  include ::Msf::Post::Windows::Registry
+
+  def initialize(info = {})
+    super
+    register_advanced_options(
+      [
+        OptInt.new('Dotnet::Post::timeout', [true, 'Dotnet execution timeout, set < 0 to run async without termination', 15]),
+        OptBool.new('Dotnet::Post::log_output', [true, 'Write output to log file', false]),
+        OptBool.new('Dotnet::Post::dry_run', [true, 'Return encoded output to caller', false]),
+        OptBool.new('Dotnet::Post::force_wow64', [true, 'Force WOW64 execution', false])
+      ], 
+      self.class
+    )
+  end
+  #
+  # Searches the subkey for the value 'Version' which contains the
+  # actual version, rather than the over-arching release
+  # An alternative would be to query for it, and catch the exception.
+  #
+  
+  def search_for_version(dotnet_subkey)
+    dotnet_version = nil
+    begin
+      subkeys = registry_enumvals(dotnet_subkey)
+    rescue ::Exception => e
```

Comment:
Generally catching ::Exception isn't the most ideal practice.  For post exploitation API, you probably just need to catch ```Rex::Post::Meterpreter::RequestError```, but I could be wrong, there might be more.

If ::Exception is the only suitable choice, then it's better to also ```elog``` the exception for future debugging purposes. For example:

```ruby
elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
```